### ISSUE TITLE
torsocks rejects connection

### ISSUE THREAD

**Issue by genodeftest** [EtiennePerot/parcimonie.sh#8]
When running parcimonie.sh I get this output:
```$ parcimonie.sh 
Sleeping 3539 seconds before refreshing key 1234567890ABCDEF1234567890ABCDEF01234567...
gpg: requesting key 0x90ABCDEF01234567 from hkps server hkps.pool.sks-keyservers.net
[Jan 05 17:33:23] WARNING torsocks[1121]: [syscall] Unsupported syscall number 186. Denying the call (in tsocks_syscall() at syscall.c:165)
[Jan 05 17:33:23] WARNING torsocks[1121]: [connect] Connection to a local address are denied since it might be a TCP DNS query to a local DNS server. Rejecting it for safety reasons. (in tsocks_connect() at connect.c:177)
gpgkeys: HTTP fetch error 7: 
gpg: no valid OpenPGP data found.
gpg: Total number processed: 0
> Sleeping 5850 seconds before refreshing key 1234567890ABCDEF1234567890ABCDEF01234568...
```(I replaced the key ids.) I am running the latest version of parcimonie.sh on torsocks 2.0.0 on Fedora 21 with no additional arguments or env variables. Any Idea what I am missing?

**Comment by genodeftest** [comment]
This is definitely not a bug in parcimonie.sh. Running strace on gpg shows me the error is in or beyond gpg. Thanks for your help and sorry for bothering you.

**Comment by EtiennePerot** [comment]
It'd still be great if you find out what the root cause was. It could be useful to someone who has the same problem and searches for this issue. Also, if it's caused by some condition that `parcimonie.sh` could detect ahead of time and warn about, such a check could be implemented.

**Comment by genodeftest** [comment]
One thing I did wrong was configuring tor as socks proxy in ~/.gnupg/gpg.conf . I effectively tried to run two layers of tor. This caused the "HTTP fetch error 7" (by curl/libcurl).
The problem still persists after fixing this.

With a .onion keyserver for both gpg and gpg2
`gpg --keyserver-options debug --recv-keys` works fine
`torsocks gpg --keyserver-options debug,http-proxy= --recv-keys` throws a warning, but finishes its job. The warning is `WARNING torsocks[25037]: [syscall] Unsupported syscall number 186. Denying the call (in tsocks_syscall() at syscall.c:165)`

When changing the keyserver to e.g. hkps://hkps.pool.sks-keyservers.net, gpg fails with error message `gpgkeys: HTTP fetch error 77:` after being connected to the server but before a SSL connection is set up. gpg2 works fine though (as above). After a while I found out that gpg does not support `~/.gnupg` in the path name for the CA certificate (keyserver-option ca-cert-file) but gpg2 does. The "HTTP fetch error 77" is caused by curl. This is the reason why gpg2 works and gpg fails. Setting an absolute path works for both gpg and gpg2.

So now only the warning about an unsupported syscall is present. This should be fixed in commit f97b07ef921dbcaef8cb6718179bce21b4714e77 in torsocks, see https://gitweb.torproject.org/torsocks.git/commit/?id=f97b07ef921dbcaef8cb6718179bce21b4714e77 for details.
